packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -380,57 +411,49 @@ class UpdatePackagesCommand extends FlutterCommand {
+    final File fakePackage = _pubspecFor(syntheticPackageDir);
+    fakePackage.createSync(recursive: true);
+    fakePackage.writeAsStringSync(
+      generateFakePubspec(
+        dependencies,
+        doUpgrade: doUpgrade,
+      ),
+    );
+    // Create a synthetic flutter SDK so that transitive flutter SDK
+    // constraints are not affected by this upgrade.
+    if (doUpgrade) {
+      temporaryFlutterSdk = createTemporaryFlutterSdk(
+        globals.logger,
+        globals.fs,
+        globals.fs.directory(Cache.flutterRoot),
+        pubspecs,
+        tempDir,
       );
-      // Create a synthetic flutter SDK so that transitive flutter SDK
-      // constraints are not affected by this upgrade.
-      if (doUpgrade) {
-        temporaryFlutterSdk = createTemporaryFlutterSdk(
-          globals.logger,
-          globals.fs,
-          globals.fs.directory(Cache.flutterRoot),
-          pubspecs,
-        );
-      }
+    }
 
-      // Next we run "pub get" on it in order to force the download of any
-      // needed packages to the pub cache, upgrading if requested.
-      await pub.get(
+    // Next we run "pub get" on it in order to force the download of any
+    // needed packages to the pub cache, upgrading if requested.
+    await pub.get(
+      context: PubContext.updatePackages,
+      project: FlutterProject.fromDirectory(syntheticPackageDir),
+      upgrade: doUpgrade,
+      offline: boolArgDeprecated('offline'),
+      flutterRootOverride: temporaryFlutterSdk?.path,
+    );
+
+    if (doUpgrade) {
+      // If upgrading, we run "pub deps --style=compact" on the result. We
+      // pipe all the output to tree.fill(), which parses it so that it can
+      // create a graph of all the dependencies so that we can figure out the
+      // transitive dependencies later. It also remembers which version was
+      // selected for each package.
+      await pub.batch(
+        <String>['deps', '--style=compact'],
         context: PubContext.updatePackages,
-        project: FlutterProject.fromDirectory(tempDir),
-        upgrade: doUpgrade,
-        offline: boolArgDeprecated('offline'),
-        flutterRootOverride: temporaryFlutterSdk?.path,
+        directory: syntheticPackageDir.path,
+        filter: tree.fill,
       );
-
-      if (doUpgrade) {
-        // If upgrading, we run "pub deps --style=compact" on the result. We
-        // pipe all the output to tree.fill(), which parses it so that it can
-        // create a graph of all the dependencies so that we can figure out the
-        // transitive dependencies later. It also remembers which version was
-        // selected for each package.
-        await pub.batch(
-          <String>['deps', '--style=compact'],
-          context: PubContext.updatePackages,
-          directory: tempDir.path,
-          filter: tree.fill,
-        );
-      }
-    } finally {
-      // Cleanup the temporary SDK
-      try {
-        temporaryFlutterSdk?.deleteSync(recursive: true);
```

Comment:
Do we ever clean up/delete this in the new code?

Reply:
Yes, right after we return from this method, we delete the parent (assuming we weren't passed an explicit `--synthetic-package-path`, which means we want it to persist after the tool exits): https://github.com/flutter/flutter/pull/115665/files#diff-61e5e832b7178fde4befd707ba407f598be7966a1671db8dc1a607661f09cca2R258